mod-di-converter-storage-server/src/main/java/org/folio/dao/association/CommonProfileAssociationDao.java

```diff
@@ -143,19 +141,22 @@ public Future<Boolean> delete(String id, ProfileSnapshotWrapper.ContentType mast
 
   @Override
   public Future<Boolean> delete(String masterWrapperId, String detailWrapperId, ProfileSnapshotWrapper.ContentType masterType,
-                                ProfileSnapshotWrapper.ContentType detailType, String jobProfileId, ReactToType reactTo, String tenantId) {
+                                ProfileSnapshotWrapper.ContentType detailType, String jobProfileId, ReactToType reactTo, Integer order, String tenantId) {
     Promise<RowSet<Row>> promise = Promise.promise();
     try {
       CQLWrapper filter = getCQLWrapper(getAssociationTableName(masterType, detailType),
-        MASTER_WRAPPER_ID_FIELD + "==" + masterWrapperId + " AND " + DETAIL_WRAPPER_ID_FIELD + "==" + detailWrapperId +
-          // if jobProfileId field defined in jsonb - perform matching on it, if not - match all records.
-          " AND (" + JOB_PROFILE_ID_FIELD + "==" + jobProfileId + " OR (cql.allRecords=1 NOT " + JOB_PROFILE_ID_FIELD + "=\"\"))");
+        MASTER_WRAPPER_ID_FIELD + "==" + masterWrapperId + " AND " + DETAIL_WRAPPER_ID_FIELD + "==" + detailWrapperId
+          + " AND (order == " + (order == null ? 0 : order) + ")"
+          + " AND (" + JOB_PROFILE_ID_FIELD + "==" + jobProfileId + " OR (cql.allRecords=1 NOT " + JOB_PROFILE_ID_FIELD + "=\"\"))");
       if (reactTo != null) {
-        filter.setWhereClause(String.format(CRITERIA_BY_REACT_TO_WHERE_CLAUSE, getAssociationTableName(masterType, detailType), reactTo.value()));
+        String whereClause = filter.getWhereClause()
+          + " AND " + String.format(CRITERIA_BY_REACT_TO_CLAUSE, getAssociationTableName(masterType, detailType), reactTo.value());
```

Comment:
do we still need the react_to clause if there is a filter by the order of profile?

Reply:
Yes, because we have the ability to add two identical action profiles to the match/don't match section (even though it doesn't make sense).